### ISSUE TITLE
pip install tflite-support raise problem with attribute 'ImageDataFree'

### ISSUE THREAD

**Issue by febrifahmi** [tensorflow/tflite-support#882]
I tried to install tflite-support with `pip install tflite-support` then run my object detection script which use vision task in tflite_support, but it's break and display `AttributeError: module 'tensorflow_lite_support.python.task.vision.core.pybinds.image_utils' has no attribute 'ImageDataFree'`.

After checking the code in `/tensorflow_lite_support/python/task/vision/core/tensor_image.py` it seems that the latest version in tflite-support's Github repo is no longer use `ImageDataFree` but use `image_data_free` instead, but somehow the package in pip ImageDataFree is still remain in the code. Hopefully the tflite-support package in pip could be updated so it reflects the latest and working version.

Thanks.

**Comment by moults31** [comment]
Same issue here, running on Raspberry Pi 4. I suspect it's fixed in `master` by https://github.com/tensorflow/tflite-support/commit/c478cf68bf9b8254e0a21b52ed3ec3c33979cff8. However, like you mention, there seems to be no way to get an Arm binary that includes this commit. I've spent hours trying to cross compile, trying to build directly on arm, etc, but no luck.

I have opened https://github.com/tensorflow/tflite-support/issues/879 for trying to build such an image from source.
I have also opened https://github.com/tensorflow/tflite-support/issues/880 for trying to get such an image as a nightly release.

Hopefully someone can come along and help us out!

**Comment by febrifahmi** [comment]
I tried to change `image_utils.ImageDataFree(self._image_data)` to `(self._image_data)` in `tensor_image.py` and it could run. However, I got `double free or corruption (!prev)` after that which I guess is unrelated to this issue.

**Comment by moults31** [comment]
Agreed that it's not likely an Arm-only issue; it's just that on Arm I have no workaround since I can't build from source.

Can you try
```
pip uninstall tflite-support
pip install tflite-support-nightly
```
That way we can know if the bleeding edge version has a fix for these issues. I can't do the above since nightly does not provide binaries for Arm anymore.

**Comment by febrifahmi** [comment]
It seems `tflite-support-nightly` still has ImageDataFree in it as shown in this error: `AttributeError: module tensorflow_lite_support.python.task.vision.core.pybinds.image_utils' has no attribute 'ImageDataFree'`. 

But somehow (thanks for your suggestion) after installing this nightly version `double free or corruption (!prev)` error disappear. While it still has `Exception ignored in: <function TensorImage.__del__ at 0x7fa9efc02280>` message, I can run my object detection scripts to check on all data that I have now.

**Comment by ncvescera** [comment]
I have the same problem, but installing the nightly version of tflite-support did not fix the issue.
I'm on a Raspberry Pi 3 and this is the error:

```
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
Exception ignored in: <function TensorImage.__del__ at 0x7f7d0263a0>
Traceback (most recent call last):
  File "/home/pi/.local/lib/python3.9/site-packages/tensorflow_lite_support/python/task/vision/core/tensor_image.py", line 105, in __del__
AttributeError: module 'tensorflow_lite_support.python.task.vision.core.pybinds.image_utils' has no attribute 'ImageDataFree'
```

**Comment by Flipajs** [comment]
For those who need a temporary solution until it is fixed, previous version work `pip install tflite-support==0.4.1`.

**Comment by febrifahmi** [comment]
I also open issue in [here](https://github.com/tensorflow/tensorflow/issues/57947) since `pip install tflite-support==0.4.1` doesn't work in my pi. I try to build tflite-support using Bazel on arm 4.2.2 but also failed, It always stop in the process. In the meantime [I trried to bootstrap from source Bazel 5.3.0](https://github.com/bazelbuild/bazel/issues/16416) on my pi in the hope of successfully build tflite-support pip wheel file but the build process also always killed in the process.

Still looking for another workaround.

**Comment by lu-wang-g** [comment]
We just fixed the [nightly Pypi](https://pypi.org/project/tflite-support-nightly/0.4.0.dev20221015/#files) a while ago for Linux / Mac / Raspberry Pi. Windows will come out later today. Let me know if it works for you. Feel free to reopen this issue as needed.

FYI, the nightly version number right now is incorrectly tagged to 0.4.0, because the package actually contains our latest changes on top of 0.4.2. We've fixed this issue.

**Comment by jhguay** [comment]
Perfect!  No error.  Thanx!




________________________________
De : lu-wang-g ***@***.***>
Envoyé : 15 octobre 2022 19:58
À : tensorflow/tflite-support ***@***.***>
Cc : Jean-Herman Guay ***@***.***>; Comment ***@***.***>
Objet : Re: [tensorflow/tflite-support] pip install tflite-support raise problem with attribute 'ImageDataFree' (Issue #882)


Closed #882<https://can01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Ftensorflow%2Ftflite-support%2Fissues%2F882&data=05%7C01%7Cjean-herman.guay%40usherbrooke.ca%7C7059c05b718e40ef78c708daaf0917ee%7C3a5a8744593545f99423b32c3a5de082%7C0%7C0%7C638014750833742297%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=NssGers8MsfuTgfhpGJxYJcoZ9bikV9tRUG0b%2Ff9sDI%3D&reserved=0> as completed.
